modules/ossm-migrating-gateways-canary.adoc

```diff
@@ -0,0 +1,87 @@
+
+// Module included in the following assemblies:
+//
+// * service-mesh-docs-main/migrating/migrating-gateways/ossm-migrating-gateways-assembly.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="ossm-migrating-gateways-canary_{context}"]
+= Gateway canary migration
+
+Use the gateway canary migration method when you want a gradual rollout. It uses multiple gateway versions with maximum control over your gateway rollout.
+
+[NOTE]
+====
+The label for the gateway namespace differs between mulitenant and cluster-wide meshes. To understand which labels to use in your specific migration case and the `maistra.io/ignore-namespace: "true"` label requirement, see the "Multitenant migration guide" or the "Cluster-wide migration guide" section.
+====
+
+.Procedure
+
+. Run the following command to label the gateway namespace to ensure that gateway injection is enabled from the new mesh and add the `maistra.io/ignore-namespace: "true"` label:
++
+[source,terminal]
+----
+$ oc label namespace <gateway_namespace> istio.io/rev=<istio_revision_name> maistra.io/ignore-namespace="true"
+----
++
+.. Remove the `istio-injection=enabled` label, if needed.
+
+. Deploy a canary gateway by using the following example:
++
+.Example YAML for canary gateway
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: istio-ingressgateway-canary
+  namespace: istio-ingress <1>
+  spec:
+     selector:
+       matchLabels:
+         istio: ingressgateway <2>
+     template:
+       metadata:
+         annotations:
+           inject.istio.io/templates: gateway
+         labels:
+           istio: ingressgateway
+           istio.io/rev: canary <3>
+       spec:
+         containers:
+         - name: istio-proxy
+           image: auto
+----
+<1> Your `Deployment` resource must be in in the same namespace as your existing gateway.
+<2> Your `spec.selector.matchlabels.istio` must match your existing gateway service selector.
```

Comment:
This should have a noun after the monospace text for clarity, I'm guessing this should be parameter:
```suggestion
<2> Your `spec.selector.matchlabels.istio` parameter must match your existing gateway service selector.
```